sorc/global_cycle.fd/cycle.f90

```diff
@@ -970,6 +1038,285 @@ SUBROUTINE ADJUST_NSST(RLA,RLO,SLMSK_TILE,SLMSK_FG_TILE,SKINT_TILE,&
+! FOUR GSI/GAUSSIAN POINTS.  DOES NOT ACCOUNT FOR MASK.  THAT
+! HAPPENS LATER.
+!----------------------------------------------------------------------
+
+ CALL REMAP_COEF( 1, IDIM, 1, JDIM, IDIM_GAUS, JDIM_GAUS, &
+                  LONS_RAD, LATS_RAD, ID1, ID2, JDC, S2C, AGRID )
+
+ DEALLOCATE(LONS_RAD, LATS_RAD, AGRID)
+!
+! Initialize variables for counts statitics to be zeros
+!
+
+! 
+ nother = 0 ! grid cells not land
+ nsnowupd = 0  ! grid cells with snow (temperature not yet updated) 
+ nsnowchange = 0  ! grid cells where no temp upd made, because snow occurence changed 
+ nnosoilnear = 0 ! grid cells where model has soil, but 4 closest gaus grids don't 
+                 ! (no update made here)
+ nsoilupd = 0 
+
+! do-will skipping above skip close-by grid cells, due to model grid arrangement?
+ 
+ IJ_LOOP : DO IJ = 1, LENSFC
+
+! for now,  do not make a temperature update if snow differs  
+! between FG and ANAL (allow correction of snow to 
+! address temperature error first) 
+
+
+   MASK_TILE    = SOILSNOW_TILE(IJ)
+   MASK_FG_TILE = SOILSNOW_FG_TILE(IJ)
+
+!----------------------------------------------------------------------
+! MASK: 1  - soil, 2 - snow, 0 - neither
+!----------------------------------------------------------------------
+
+   IF (MASK_TILE == 0) THEN ! skip if neither soil nor snow
+     nother = nother + 1
+     CYCLE IJ_LOOP  
+   ENDIF
+
+
+!  Get i,j index on array of (idim,jdim) from known ij
+
+   JTILE = (IJ-1) / IDIM + 1
+   ITILE = MOD(IJ,IDIM)
+   IF (ITILE==0) ITILE = IDIM
+
+!----------------------------------------------------------------------
+! if the snow analysis has chnaged to occurence of snow, skip the 
+! temperature analysis
+!----------------------------------------------------------------------
+
+   IF ((MASK_FG_TILE == 2 .AND. MASK_TILE == 1) .or. (MASK_FG_TILE == 1 .AND. MASK_TILE == 2) ) THEN
+     nsnowchange = nsnowchange + 1
+     CYCLE IJ_LOOP  
+   ENDIF
+
+!----------------------------------------------------------------------
+!  DO UPDATE TO SOIL TEMPERATURE GRID CELLS, USING BILINEAR INTERP 
+!----------------------------------------------------------------------
+ 
+   IF (MASK_TILE == 1) THEN
+       ! these are the four nearest grid cells on the GAUS grid
+       IGAUS   = ID1(ITILE,JTILE)
+       JGAUS   = JDC(ITILE,JTILE)
+       IGAUSP1 = ID2(ITILE,JTILE)
+       JGAUSP1 = JDC(ITILE,JTILE)+1
+  
+! MAKE SURE GAUS GRID HAS SOIL NEARBY 
+       gaus_has_soil = .false.
+       IF (SOILSNOW_GAUS(IGAUS,JGAUS)     == 1 .OR. &
+           SOILSNOW_GAUS(IGAUSP1,JGAUS)   == 1 .OR. &
+           SOILSNOW_GAUS(IGAUSP1,JGAUSP1) == 1 .OR. &
+           SOILSNOW_GAUS(IGAUS,JGAUSP1)   == 1)  gaus_has_soil = .true. 
+       
+       IF (.not. gaus_has_soil) THEN
+         nnosoilnear = nnosoilnear + 1 
+         CYCLE IJ_LOOP 
+       ENDIF
+
+! CALCUALATE WEIGHTED INCREMENT OVER NEARBY GRID CELLS THAT HAVE SOIL
+
+! DRAPER: TO-DO, CODE ADDING INCREMENTS TO SOIL MOISTURE. 
```

Comment:
After the increment is applied, how much will the soil temperature change?  Is the adjustment larger in the upper or lower layers?

Reply:
The increments are generated by an EnKF, so the exact pattern depends on the EnKF inputs, but in general they decrease rapidly with depth, and are fairly small in an individual time step (< 0.5 K at the most in the surface layer). The soil EnKF is still being refined, and one of the issues I need to look at is whether to switch off the increments in the deepest layers.

Reply:
Ok.  Any change in soil temperature will change the percentage of frozen soil moisture.  A change <0.5K for cold layers may be small enough to not require an adjustment.  If a layer rises above freezing, the liquid soil moisture could be set to the total soil moisture.  @jiaruidong2017 do you have an opinion?

Reply:
Good catch, George. For now, I'm not creating increments near freezing (or when the atmosphere is near freezing), but I think we should address this anyway.

Reply:
@HelinWei-NOAA may want to comment as well.

Reply:
An issue here is that we're going to have to pull in model-specific code to make these consistency corrections to other variables after the land DA update (the same issue is going to come up when we add the snow increments). Does anyone have a good idea for how to handle this? I could write Noah model-specific consistency routines, and then later Noah-MP specific consistency routines. Other land DA systems I've worked with call routines along the lines of 'if soil temperature updated, check if soil liquid/ice needs to be updated' How do we feel about this code being inside UFS_UTILS/global_cycle though?  Is global_cycle already Noah-specific? @barlage